src/platform/Linux/PlatformManagerImpl.cpp

```diff
@@ -187,6 +278,59 @@ CHIP_ERROR PlatformManagerImpl::_GetCurrentHeapHighWatermark(uint64_t & currentH
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR PlatformManagerImpl::_GetRebootCount(uint16_t & rebootCount)
+{
+    uint32_t count = 0;
+
+    CHIP_ERROR err = ConfigurationMgrImpl().GetRebootCount(count);
+
+    if (err == CHIP_NO_ERROR)
+        rebootCount = static_cast<uint16_t>(count);
+
+    return err;
+}
+
+CHIP_ERROR PlatformManagerImpl::_GetUpTime(uint64_t & upTime)
+{
+    uint64_t currentTimeMilliseconds = System::Clock::GetMonotonicMilliseconds();
+
+    if (currentTimeMilliseconds >= mStartTimeMilliseconds)
+    {
+        upTime = (currentTimeMilliseconds - mStartTimeMilliseconds) / 1000;
+        return CHIP_NO_ERROR;
+    }
+
+    return CHIP_ERROR_INVALID_TIME;
+}
+
+CHIP_ERROR PlatformManagerImpl::_GetTotalOperationalHours(uint32_t & totalOperationalHours)
+{
+    uint64_t upTime = 0;
+
+    if (_GetUpTime(upTime) == CHIP_NO_ERROR)
+    {
+        uint32_t totalHours = 0;
+        if (ConfigurationMgrImpl().GetTotalOperationalHours(totalHours) == CHIP_NO_ERROR)
+        {
+            totalOperationalHours = totalHours + static_cast<uint32_t>(upTime / 3600);
+        }
+    }
+
+    return CHIP_ERROR_INVALID_TIME;
+}
+
+CHIP_ERROR PlatformManagerImpl::_GetBootReasons(uint8_t & bootReasons)
+{
+    uint32_t reason = 0;
+
+    CHIP_ERROR err = ConfigurationMgrImpl().GetBootReasons(reason);
+
+    if (err == CHIP_NO_ERROR)
+        bootReasons = static_cast<uint8_t>(reason);
```

Comment:
Similar here: if we just want an 8-bit bitfield, why are we storing/retrieving a 32-bit int?

Reply:
Same above, we don't have low level API     static CHIP_ERROR ReadConfigValue(Key key, uint8_t & val); to read out 8-bit value